Author: Saleem Abdulrasool
Upstream: no
Reason: dont build all the extras, only gudev

Edited by Daniel Mierswa <impulze@impulze.org> for udev-153.

diff -Naur udev-153/configure.ac udev-153/configure.ac
--- udev-153/configure.ac	2010-04-21 08:30:43.000000000 +0200
+++ udev-153/configure.ac	2010-05-02 17:12:16.000000000 +0200
@@ -46,6 +46,36 @@
 	AC_DEFINE(ENABLE_LOGGING, [1], [System logging.])
 fi
 
+AC_ARG_ENABLE([gudev],
+              AS_HELP_STRING([--disable-gudev], [disable libgudev - libudev gobject interface]),
+	      [], [enable_gudev=yes])
+if test x"$enable_gudev" = x"yes" ; then
+    PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.7.0 gobject-2.0 >= 2.7.0])
+    AC_SUBST([GLIB_CFLAGS])
+    AC_SUBST([GLIB_LIBS])
+fi
+AM_CONDITIONAL([ENABLE_GUDEV], [test x"$enable_gudev" = x"yes"])
+
+AC_ARG_WITH(firmware-path,
+	AS_HELP_STRING([--with-firmware-path=DIR[[[:DIR[...]]]]],
+	   [Firmware search path (default=/lib/firmware/updates:/lib/firmware)]),
+	[],
+	[with_firmware_path="/lib/firmware/updates:/lib/firmware"]
+	)
+OLD_IFS=$IFS
+IFS=:
+for i in $with_firmware_path
+do
+	if test "x${FIRMWARE_PATH}" = "x"
+	then
+		FIRMWARE_PATH="\\\"${i}/\\\""
+	else
+		FIRMWARE_PATH="${FIRMWARE_PATH}, \\\"${i}/\\\""
+	fi
+done
+IFS=$OLD_IFS
+AC_SUBST([FIRMWARE_PATH], [$FIRMWARE_PATH])
+
 AC_ARG_ENABLE([extras],
 	AS_HELP_STRING([--disable-extras], [disable extras with external dependencies]),
 	[], [enable_extras=yes])
@@ -55,10 +85,6 @@
 		AC_MSG_ERROR([gperf is needed])
 	fi
 
-	PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.7.0 gobject-2.0 >= 2.7.0])
-	AC_SUBST([GLIB_CFLAGS])
-	AC_SUBST([GLIB_LIBS])
-
 	AC_CHECK_LIB([acl], [acl_init], [:], AC_MSG_ERROR([libacl not found]))
 	AC_CHECK_HEADER([acl/libacl.h], [:], AC_MSG_ERROR([libacl header not found]))
 
@@ -82,26 +108,6 @@
 		fi])
 	AC_SUBST(PCI_DATABASE)
 
-	AC_ARG_WITH(firmware-path,
-		AS_HELP_STRING([--with-firmware-path=DIR[[[:DIR[...]]]]],
-		   [Firmware search path (default=/lib/firmware/updates:/lib/firmware)]),
-		[],
-		[with_firmware_path="/lib/firmware/updates:/lib/firmware"]
-		)
-	OLD_IFS=$IFS
-	IFS=:
-	for i in $with_firmware_path
-	do
-		if test "x${FIRMWARE_PATH}" = "x"
-		then
-			FIRMWARE_PATH="\\\"${i}/\\\""
-		else
-			FIRMWARE_PATH="${FIRMWARE_PATH}, \\\"${i}/\\\""
-		fi
-	done
-	IFS=$OLD_IFS
-	AC_SUBST([FIRMWARE_PATH], [$FIRMWARE_PATH])
-
 	AC_CHECK_HEADER([linux/input.h], [:], AC_MSG_ERROR([kernel headers not found]))
 	AC_SUBST([INCLUDE_PREFIX], [$(echo '#include <linux/input.h>' | eval $ac_cpp -E - | sed -n '/linux\/input.h/ {s:.*"\(.*\)/linux/input.h".*:\1:; p; q}')])
 fi
diff -Naur udev-153/Makefile.am udev-153/Makefile.am
--- udev-153/Makefile.am	2010-04-20 07:02:58.000000000 +0200
+++ udev-153/Makefile.am	2010-05-02 17:11:36.000000000 +0200
@@ -285,15 +285,9 @@
 libexec_PROGRAMS += extras/v4l_id/v4l_id
 dist_udevrules_DATA += extras/v4l_id/60-persistent-v4l.rules
 
-if ENABLE_EXTRAS
-# ------------------------------------------------------------------------------
-# conditional extras (need glib, libusb, libacl, ...)
-# ------------------------------------------------------------------------------
-dist_udevrules_DATA += \
-	rules/rules.d/75-net-description.rules \
-	rules/rules.d/75-tty-description.rules \
-	rules/rules.d/78-sound-card.rules
+BUILT_SOURCES =
 
+if ENABLE_GUDEV
 # ------------------------------------------------------------------------------
 # GUdev - libudev gobject interface
 # ------------------------------------------------------------------------------
@@ -331,7 +325,7 @@
 	extras/gudev/gudevmarshal.c \
 	extras/gudev/gudevenumtypes.h \
 	extras/gudev/gudevenumtypes.c
-BUILT_SOURCES = $(dist_extras_gudev_libgudev_1_0_la_SOURCES)
+BUILT_SOURCES += $(dist_extras_gudev_libgudev_1_0_la_SOURCES)
 
 extras_gudev_libgudev_1_0_la_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
@@ -408,6 +402,17 @@
 CLEANFILES += $(gir_DATA) $(typelibs_DATA)
 endif # ENABLE_INTROSPECTION
 
+endif # ENABLE_GUDEV
+
+if ENABLE_EXTRAS
+# ------------------------------------------------------------------------------
+# conditional extras (need libusb, libacl, ...)
+# ------------------------------------------------------------------------------
+dist_udevrules_DATA += \
+	rules/rules.d/75-net-description.rules \
+	rules/rules.d/75-tty-description.rules \
+	rules/rules.d/78-sound-card.rules
+
 # ------------------------------------------------------------------------------
 # Bluetooth HID devices with special magic to switch the device
 # ------------------------------------------------------------------------------
